pkg/tools/tfe/get_tf_state_and_config.go

```diff
@@ -0,0 +1,267 @@
+	if err != nil {
+		return nil, utils.LogAndReturnError(logger, "reading workspace details", err)
+	}
+
+	// Convert workspace to JSON-serializable format
+	workspaceInfo := &client.WorkspaceInfo{
+		ID:                   workspace.ID,
+		Name:                 workspace.Name,
+		Description:          workspace.Description,
+		Environment:          workspace.Environment,
+		AutoApply:            workspace.AutoApply,
+		TerraformVersion:     workspace.TerraformVersion,
+		WorkingDirectory:     workspace.WorkingDirectory,
+		ExecutionMode:        string(workspace.ExecutionMode),
+		ResourceCount:        workspace.ResourceCount,
+		ApplyDurationAverage: int64(workspace.ApplyDurationAverage),
+		PlanDurationAverage:  int64(workspace.PlanDurationAverage),
+	}
+
+	// Build response with metadata using JSON-friendly struct
+	response := &client.StateAndConfigJSONResponse{
+		Type:      "get_tf_state_and_config",
+		Success:   true,
+		Workspace: workspaceInfo,
+		Metadata: &client.ResponseMetadata{
+			RetrievedAt:      time.Now(),
+			WorkspaceID:      workspace.ID,
+			OrganizationName: terraformOrgName,
+			WorkspaceName:    workspaceName,
+		},
+	}
+
+	// Fetch state data
+	stateContent, err := fetchStateContent(ctx, tfeClient, workspace.ID, logger)
+	if err != nil {
+		logger.WithError(err).Warn("failed to fetch state data, continuing without it")
+	} else {
+		response.TfStateFileContent = stateContent
+	}
+
+	// Fetch configuration data
+	configData, err := fetchConfigurationData(ctx, tfeClient, workspace, logger)
+	if err != nil {
+		logger.WithError(err).Warn("failed to fetch configuration data, continuing without it")
+	} else {
+		response.ConfigData = configData
+	}
+
+	// Debug: Log what we're about to return
+	if response.TfStateFileContent != nil {
+		logger.WithFields(log.Fields{
+			"tf_state_file_content_keys": getMapKeys(response.TfStateFileContent),
+			"has_tf_state_file_content":  response.TfStateFileContent != nil,
+		}).Info("Response state content summary")
+	}
+
+	// Debug: Log response preparation
+	logger.WithFields(log.Fields{
+		"has_tf_state_content": response.TfStateFileContent != nil,
+		"has_config_data":      response.ConfigData != nil,
+		"workspace_id":         response.Workspace.ID,
+	}).Info("Final structured response prepared")
+
+	return mcp.NewToolResultStructuredOnly(response), nil
+}
+
+// fetchStateContent retrieves and parses the complete Terraform state file
+func fetchStateContent(ctx context.Context, tfeClient *tfe.Client, workspaceID string, logger *log.Logger) (map[string]interface{}, error) {
+	// Get current state version
+	stateVersion, err := tfeClient.StateVersions.ReadCurrent(ctx, workspaceID)
+	if err != nil {
+		return nil, utils.LogAndReturnError(logger, "reading current state version", err)
+	}
+
+	// Download and parse the state file for raw content
+	if stateVersion.JSONDownloadURL != "" {
+		stateFileContent, err := downloadStateFile(ctx, stateVersion.JSONDownloadURL, logger)
```

Comment:
statefile can contain some sensitive info, should we return all of it? I mean maybe but a skimmed version of it